From b2911db973f680c874eb76c330e493584afe88f9 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Mon, 13 Nov 2017 02:48:36 +0000
Subject: [PATCH 3/3] cmd/snap-confine: Add missing bi-arch NVIDIA files

Additionally, let's ensure we have parity with the NVIDIA bind-mounts on
multiarch by ensuring that our vdpau driver for NVIDIA is also accessible
under the rootfs.

To make VDPAU accessible, one must set `VDPAU_DRIVER_PATH` to include
the `/var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl/vdpau` paths within
the snap to ensure that vdpau drivers for both biarch and multiarch hosts
can be found trivially.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 cmd/snap-confine/mount-support-nvidia.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/cmd/snap-confine/mount-support-nvidia.c b/cmd/snap-confine/mount-support-nvidia.c
index 29522dc45..b6e45cf76 100644
--- a/cmd/snap-confine/mount-support-nvidia.c
+++ b/cmd/snap-confine/mount-support-nvidia.c
@@ -82,6 +82,7 @@ static const char *nvidia_globs[] = {
 	"/usr/lib/libnvidia-cfg.so*",
 	"/usr/lib/libnvidia-compiler.so*",
 	"/usr/lib/libnvidia-eglcore.so*",
+	"/usr/lib/libnvidia-egl-wayland*",
 	"/usr/lib/libnvidia-encode.so*",
 	"/usr/lib/libnvidia-fatbinaryloader.so*",
 	"/usr/lib/libnvidia-fbc.so*",
@@ -91,6 +92,7 @@ static const char *nvidia_globs[] = {
 	"/usr/lib/libnvidia-ml.so*",
 	"/usr/lib/libnvidia-ptxjitcompiler.so*",
 	"/usr/lib/libnvidia-tls.so*",
+	"/usr/lib/vdpau/libvdpau_nvidia.so*",
 };
 
 static const size_t nvidia_globs_len =
@@ -127,6 +129,7 @@ static const char *nvidia_globs32[] = {
 	"/usr/lib32/libnvidia-ml.so*",
 	"/usr/lib32/libnvidia-ptxjitcompiler.so*",
 	"/usr/lib32/libnvidia-tls.so*",
+	"/usr/lib32/vdpau/libvdpau_nvidia.so*",
 };
 
 static const size_t nvidia_globs32_len =
-- 
2.15.0

